background.js

```diff
@@ -1,28 +1,36 @@
 // triggered when user clicks on installed extention icon
-chrome.browserAction.onClicked.addListener(function(tab) {
-    
-    // retrive domain from active tab
-    var url = tab.url;
-    var matches = url.match(/^https?\:\/\/([^\/?#]+)(?:[\/?#]|$)/i);
-    var d = matches && matches[1].replace('www.','');
-    d = '.'+d;
+chrome.browserAction.onClicked.addListener(function (tab) {
 
-    // get all cookies for domain
-    chrome.cookies.getAll({domain: d}, function (cookies) {
+    let url = tab.url;
 
-        // iterate on cookie to get cookie detail
-        for (var i=0; i<cookies.length; i++) {
-            var url = "http" + (cookies[i].secure ? "s" : "") + "://" + cookies[i].domain + cookies[i].path;
-            var cname = cookies[i].name;
+    // clear all cookies for the current url
+    chrome.cookies.getAll({ url: url }, function (cookies) {
```

Comment:
This is the ultimate fix - get the cookies for the current url and clear those.
I'm not sure *why* this differs from what you'd get from the entire domain.

We also clear the domain cookies a little further down

Reply:
👍